src/Storage/Storage/Common/SasTokenHelper.cs

```diff
@@ -328,7 +345,115 @@ public static ShareSignedIdentifier GetShareSignedIdentifier(ShareClient share,
+                    {
+                        throw new ArgumentException(Resources.SignedExpiryTimeMustBeOmitted);
+                    }
+                    else
+                    {
+                        sasBuilder.ExpiresOn = expiryTime.Value.ToUniversalTime();
+                    }
+                }
+                // Set up expiry time if it is not set by user input or the policy
+                else if (signedIdentifier.AccessPolicy.ExpiresOn == DateTimeOffset.MinValue || signedIdentifier.AccessPolicy.ExpiresOn == null)
+                {
+                    if (sasBuilder.StartsOn != DateTimeOffset.MinValue && sasBuilder.StartsOn != null)
+                    {
+                        sasBuilder.ExpiresOn = sasBuilder.StartsOn.ToUniversalTime().AddHours(1);
+                    }
+                    else
+                    {
+                        sasBuilder.ExpiresOn = DateTimeOffset.UtcNow.AddHours(1);
+                    }
+                }
+                if (permission != null)
+                {
+                    if (signedIdentifier.AccessPolicy.Permissions != null)
+                    {
+                        throw new ArgumentException(Resources.SignedPermissionsMustBeOmitted);
+                    }
+                    else
+                    {
+                        sasBuilder.SetPermissions(permission, true);
+                    }
+                }
+            }
+            else
+            {
+                sasBuilder.SetPermissions(permission, true);
+
+                if (startTime != null)
+                {
+                    sasBuilder.StartsOn = startTime.Value.ToUniversalTime();
+                }
+                if (expiryTime != null)
+                {
+                    sasBuilder.ExpiresOn = expiryTime.Value.ToUniversalTime();
+                }
+                else
+                {
+                    if (sasBuilder.StartsOn != DateTimeOffset.MinValue)
+                    {
+                        sasBuilder.ExpiresOn = sasBuilder.StartsOn.AddHours(1).ToUniversalTime();
+                    }
+                    else
+                    {
+                        sasBuilder.ExpiresOn = DateTimeOffset.UtcNow.AddHours(1);
+                    }
+                }
+                if (iPAddressOrRange != null)
+                {
+                    sasBuilder.IPRange = Util.SetupIPAddressOrRangeForSASTrack2(iPAddressOrRange);
+                }
+                if (protocol != null)
+                {
+                    if (protocol.Value == SharedAccessProtocol.HttpsOrHttp)
+                    {
+                        sasBuilder.Protocol = SasProtocol.HttpsAndHttp;
+                    }
+                    else //HttpsOnly
+                    {
+                        sasBuilder.Protocol = SasProtocol.Https;
+                    }
+                }
+            }
```

Comment:
iPAddressOrRange , protocol  should be move 1 level out.
No matter use policy or not, we should always check iPAddressOrRange , protocol .